### ISSUE TITLE
Increased 404s for lighthouse_worker in Canary Chrome DevTools

### ISSUE THREAD

**Issue by connorjclark** [GoogleChrome/lighthouse#10883]
It's not uncommon to download the latest Canary and see Lighthouse fail to load.

**TL;DR** [REMOTE_MODULE_FALLBACK_REVISION](https://source.chromium.org/chromium/chromium/src/+/master:third_party/devtools-frontend/src/front_end/root/Runtime.js;l=6;drc=f1f00fa32ac4bb8470de40bc506e36d13a8babbd?originalUrl=https:%2F%2Fcs.chromium.org%2F) needs to be updated.

It is common for the latest version of the `lighthouse_worker` remote module to not be built yet. This is a known issue, and we should consider updating the uploader code to prioritize the latest canary releases. When the latest version 404s, a fallback is used. However, the module name has changed, so just swapping the revision part of the URL doesn't work.

The fallback URL: https://chrome-devtools-frontend.appspot.com/serve_file/@010ddcfda246975d194964ccf20038ebbdec6084/lighthouse_worker/lighthouse_worker_module.js (404)

For that revision the worker was named audits2: https://chrome-devtools-frontend.appspot.com/serve_file/@010ddcfda246975d194964ccf20038ebbdec6084/lighthouse_worker/lighthouse_worker_module.js (200 - and Lighthouse 4!)

Action items:
* Update the fallback revision to a recent revision. This has a narrow impact because only `lighthouse_worker` is attempted a second time with a fallback.
* For every roll to devtools, follow up by bumping this fallback revision.
* Update the uploader to prioritize the latest canary releases / revisions.

____

## Notes

Open Lighthouse Panel, open devtools-on-devtools, run Lighthouse, pause before it finishes. In DoD go to Sources and open RuntimeInstantiator in the lighthouse_worker.js target. Set a breakpoint in `startWorker`. Hit play, and run Lighthouse again to trigger the breakpoint before the remote module is loaded.

https://source.chromium.org/chromium/chromium/src/+/master:third_party/devtools-frontend/src/front_end/root/Runtime.js;bpv=0;bpt=0?originalUrl=https:%2F%2Fcs.chromium.org%2F

"loadScriptsPromise" in root.js

"loadResourcePromiseWithFallback" 

"internalLoadResourcePromise"

```
root.js:124 Empty response arrived for script 'devtools://devtools/remote/serve_file/@d5e62ea5dcc3e687407cf9ea43c34f30969675f5/lighthouse_worker/lighthouse_worker_module.js'
evaluateScript @ root.js:124
scriptSourceLoaded @ root.js:123
Promise.then (async)
loadScriptsPromise @ root.js:122
_loadScripts @ root.js:71
Promise.then (async)
_loadPromise @ root.js:64
instance @ root.js:84
_dispatchMessage @ worker_service.js:7
_dispatchMessageWrapped @ worker_service.js:5
_onMessage @ worker_service.js:21
VM10:1 Uncaught (in promise) ReferenceError: LighthouseService is not defined
    at eval (eval at _createInstance (root.js:87), <anonymous>:1:1)
    at eval (<anonymous>)
    at Extension._createInstance (root.js:87)
```

Error was swallowed in "internalLoadResourcePromise", set breakpoint there.

"While loading from url devtools://devtools/remote/serve_file/@d5e62ea5dcc3e687407cf9ea43c34f30969675f5/lighthouse_worker/lighthouse_worker_module.js server responded with a status of 404"

Then fallback revision is attempted, also 404s.

Let's look into this `devtools://` protocol.

found https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/devtools/url_constants.cc;bpv=1;bpt=1;l=10?q=serve_file&ss=chromium%2Fchromium%2Fsrc:chrome%2Fbrowser%2Fdevtools%2F&originalUrl=https:%2F%2Fcs.chromium.org%2F

`devtools://` gets converts to a normal `https://` request: https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/devtools/devtools_ui_bindings.cc;l=292;drc=511af4a1ebc3a3dcdc0f9b6b684068b5c4792e26;bpv=0;bpt=1?originalUrl=https:%2F%2Fcs.chromium.org%2F

`devtools://devtools/remote/serve_file/@d5e62ea5dcc3e687407cf9ea43c34f30969675f5/lighthouse_worker/lighthouse_worker_module.js`
->
`https://chrome-devtools-frontend.appspot.com/serve_file/@d5e62ea5dcc3e687407cf9ea43c34f30969675f5/lighthouse_worker/lighthouse_worker_module.js`

`components/ui_devtools/devtools_server.cc`

**Comment by paulirish** [comment]
> TL;DR REMOTE_MODULE_FALLBACK_REVISION needs to be updated.

I was using another solution/hack because previous devtools TL wouldn't let me update that fallback revision. I updated the GAE commit metadata for the existing fallback revision. I just had it pretend like that hash was really a chrome 77 hash instead of a chrome 55 hash (or whatever).  This basically uses the "Redirection hack" as mentioned in the DevTools Uploader guide.

**Comment by paulirish** [comment]
Today we ran into this sorta thing again. the 404s were fine in aggregate but we had one notable user get one. (And he got served the [updated fallback](https://chromium-review.googlesource.com/c/devtools/devtools-frontend/+/2225600) thx to connor)

Our user updated to the latest Canary sometime this afternoon. That build (86.0.4189.0) was built from a git commit that happened at 11:24am. Chrome builders finished the Canary build at 1:18pm. (Presumably, Chrome started pushing out that update around then)  

The uploader had started a new cycle at 11:19am. It fetches latest from the repo right then. so it missed our key commit by 5 minutes.  The cycle ran for a few hours and a new one started right after 3:00pm. And at 3:06pm, the DevTools uploaded the latest version, fixing the LH version in canary.  Our user found a very special window. :)

--------

### fixes

A while ago, me and connor made sure the uploader prioritizes release-bot commits, which are the key ones (and this was one.)  https://chromium.googlesource.com/chromium/tools/chrome-devtools-frontend/+/refs/heads/master/gce/uploader_iteration.sh#140  

However since uploader cycles can take 4-8 hours, there's risk for anyone who is running an hours-old build. 

we can do a couple things

1. make the uploader cycle shorter
1. upload any key release-bot commits more frequently than once per cycle.

More medium-term, we also could reconsider how we handle network fallback. One alternative is serving Chrome 86 known resources to Chrome 86.xx  if we don't have that particular commit. That'd also fix Brave and Yandex browsers, who get a pretty broken experience. We'd need to account for the remote-debugging case, still.  

Gosh the uploader is a pain.

**Comment by connorjclark** [comment]
part relevant to #11029:

> Update the uploader to prioritize the latest canary releases / revisions.


We should do this, I think

> upload any key release-bot commits more frequently than once per cycle.
